sechub-doc/src/docs/asciidoc/sechub-getting-started/run-all-steps.sh

```diff
@@ -4,42 +4,52 @@
 SECHUB_SERVER="https://localhost:8443"
 
 # Get sechub's root folder
-currentDir=$(pwd)
-baseDir="$currentDir"
-regex=sechub$
-
-while ! [[ "$baseDir" =~ ${regex} ]]; do
-    baseDir="$(dirname "$baseDir")"
-done
+baseDir="$(git rev-parse --show-toplevel)" 
 
 # Copy all scripts from the current folder
 cp ./* "$baseDir"
 rm "$baseDir/run-all-steps.sh"
 cd "$baseDir" || exit
 
 stopServers() {
-    sh ./stop-sechub.sh "no" "./sechub-solution/docker-compose_sechub"
-    sh ./stop-pds-gosec.sh "./sechub-pds-solutions/gosec/docker-compose_pds_gosec_external-network.yaml"
-    exit 0
+    sh ./stop-sechub.sh
+    sh ./stop-pds-gosec.sh
+    exit $1
 }
 
+trap 'stopServers 0' INT
+
 # Start SecHub
-nohup "$(sh ./start-sechub.sh)" &
+nohup sh ./start-sechub.sh &
 
+# Wait until sechub is started
+TRIES=0
 while true; do
+    
     echo "Waiting for SecHub server to start.."
     isStarted="$(curl -sw '%{http_code}' -k $SECHUB_SERVER/api/anonymous/check/alive)"
     if [ "$isStarted" == "200" ]; then
         break
     fi
+
+    if [ "$TRIES" -gt 10 ]; then
```

Comment:
Please add a condition the condition to the while loop and check this condition outside the loop.
In pseudo-code:

~~~
while $tries -le xy
then
   // check if server started
done

if "$TRIES" -gt 10
then
  …
  stopServers 1
fi
~~~

This change reduces the complexity of the control flow. In addition, using the condition of the while loop makes the code easier to read.